packages/geo/src/lib/filter/ogc-filter-toggle-button/ogc-filter-toggle-button.component.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  Component,
+  Input,
+  ChangeDetectorRef,
+  OnInit
+} from '@angular/core';
+
+import {
+  OgcFilterableDataSource,
+  IgoOgcFilterObject,
+  OgcPushButton,
+  OgcPushButtonBundle
+
+} from '../../filter/shared/ogc-filter.interface';
+import { OgcFilterWriter } from '../../filter/shared/ogc-filter';
+import { IgoMap } from '../../map';
+import { OGCFilterService } from '../shared/ogc-filter.service';
+import { WMSDataSource } from '../../datasource/shared/datasources/wms-datasource';
+
+@Component({
+  selector: 'igo-ogc-filter-toggle-button',
+  templateUrl: './ogc-filter-toggle-button.component.html',
+  styleUrls: ['./ogc-filter-toggle-button.component.scss']
+})
+export class OgcFilterToggleButtonComponent implements OnInit {
+  private ogcFilterWriter: OgcFilterWriter;
+  private _dataSource: OgcFilterableDataSource;
+  public color = 'primary';
+  private _map: IgoMap;
+
+  @Input() refreshFilters: () => void;
+
+  @Input()
+  get datasource(): OgcFilterableDataSource {
```

Comment:
setters should be defined first for better results in compodoc

Reply:
done